openedx_lti_tool_plugin/signals.py

```diff
@@ -24,3 +32,36 @@ def create_course_access_configuration(
     # Only create a CourseAccessConfiguration instance if LtiTool was created.
     if created:
         CourseAccessConfiguration.objects.get_or_create(lti_tool=instance)
+
+
+@receiver(course_grade_changed())
+def update_course_score(
+    sender: Any,  # pylint: disable=unused-argument
+    user: UserT,
+    course_grade: Any,
+    course_key: CourseKey,
+    **kwargs,
+):
+    """Update score for LTI graded resources with course ID as context key.
+
+    Args:
+        sender: Signal sender argument.
+        user: User instance.
+        course_grade: Course grade object.
+        course_key: Course opaque key.
+        **kwargs: Arbitrary keyword arguments.
+    """
+    # Ignore signal if plugin is disabled, is not a LTI user grade
+    # or the course grade has not been passed.
+    if (
+        not getattr(settings, 'OLTITP_ENABLE_LTI_TOOL', False)
+        or not getattr(user, 'openedx_lti_tool_plugin_lti_profile', None)
+        or not course_grade.passed
+    ):
+        return None
+
+    # Update score for all related graded resources.
+    for graded_resource in LtiGradedResource.objects.all_from_user_id(user_id=user.id, context_key=course_key):
+        graded_resource.update_score(course_grade.percent, MAX_SCORE, datetime.now(tz=timezone.utc))
```

Comment:
Can a course have more than one graded resource? If yes, in that case wouldn't you setting the course_grade.percent to all resources?. @kuipumu

Reply:
@Jacatove A course can have more than one graded resource, a graded resource is the representation of an AGS endpoint requesting score updates, we are querying all the endpoints that are requesting a score update for this course and sending an update with the current grade to all of them